gio/src/file.rs

```diff
@@ -1111,6 +1113,171 @@ pub trait FileExtManual: IsA<File> + Sized {
 
         (fut, Box::pin(receiver))
     }
+
+    #[doc(alias = "g_file_set_attribute")]
+    fn set_attribute(
+        &self,
+        attribute: &str,
+        type_value: FileAttributeTypeValue,
+        flags: FileQueryInfoFlags,
+        cancellable: Option<&impl IsA<Cancellable>>,
+    ) -> Result<(), glib::Error> {
+        unsafe {
+            let mut error = std::ptr::null_mut();
+            let is_ok = ffi::g_file_set_attribute(
+                self.as_ref().to_glib_none().0,
+                attribute.to_glib_none().0,
+                type_value.as_ref().type_().into_glib(),
+                type_value.as_ref().value().as_ptr(),
+                flags.into_glib(),
+                cancellable.map(|p| p.as_ref()).to_glib_none().0,
+                &mut error,
+            );
+            debug_assert_eq!(is_ok == glib::ffi::GFALSE, !error.is_null());
+            if error.is_null() {
+                Ok(())
+            } else {
+                Err(from_glib_full(error))
+            }
+        }
+    }
 }
 
 impl<O: IsA<File>> FileExtManual for O {}
+
+#[derive(Debug)]
+pub struct FileAttributeTypeValue<'a>(FileAttributeTypeValueInner<'a>);
```

Comment:
```suggestion
pub struct FileAttributeValue<'a>(FileAttributeValueInner<'a>);
```

the type seems a bit redundant to me, but no strong opinion